modules/microshift-disabling-lvms-csi-snapshot.adoc

```diff
@@ -0,0 +1,45 @@
+// Module included in the following assemblies:
+//
+// * microshift_storage/microshift-storage-plugin-overview.adoc
+// * microshift_configuring/microshift-disable-lvms-csi-provider-csi-snapshot.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-disabling-lvms-csi-snapshot_{context}"]
+= Disabling deployments that run CSI snapshot implementations
+
+Before you disable the CSI snapshot implementation pods, you must manually stop any user workloads that have attached volumes. Use the following procedure to disable installation of the CSI implementation pods.
+
+[NOTE]
+====
+{microshift-short} will not delete CSI snapshot implementation pods. You must configure {microshift-short} to disable installation of the CSI snapshot implementation pods during the startup process.
+====
+
+.Procedure
+
+. Disable installation of the CSI snapshot controller by entering the `optionalCsiComponents` value under the `storage` section of the {microshift-short} configuration file:
++
+[source,yaml]
+----
+# ...
+  storage
+   optionalCsiComponents: "csi-snapshot-controller" <1>
+# ...
+----
+<1> Accepted values are `csi-snapshot-controller`, `csi-snapshot-webhook`, or `none`. `none` is mutually exclusive with all other values.
++
+[NOTE]
+====
+If the `optionalCsiComponents` value is empty or null, {microshift-short} defaults to deploying snapshot-controller and snapshot-webhook.
```

Comment:
Can you ask a developer to provide more details around empty or null ?

I tried 

`   optionalCsiComponents:
       - [""]
`

'   optionalCsiComponents:
       - ""
'

'   optionalCsiComponents:
       - ""
'
Each version fails when I try to restart microshift.service

Reply:
@copejon Please confirm

Reply:
This is dev bug. I need to confirm a solution before I can comment here.

Reply:
A fix is up, but CI his having a rough day: https://github.com/openshift/microshift/pull/3914

An empty value would be an empty array: `[]`
A null value would be an undefined value: `optionalCsiComponents: ` or simply not providing the field at all:

```
storage: {}
```

Reply:
To summarize, valid entries are:
1. Not defining the field at all, as it's optional.
2. Specifying the field with an empty value ( `[]` )
3. Specifying the field with one of the accepted values

Reply:
done